### ISSUE TITLE
Create utility method for computing new bounding box after a transform

### ISSUE THREAD

**Issue by ctrueden** [imglib/imglib2-realtransform#6]
At least for affine transforms, it is reasonable to take the corners of the source interval, and feed them into the transform to discern the boundaries of the target interval. This is the default expected behavior when performing common image processing operations such as shearing. See [this forum thread](http://forum.imagej.net/t/using-imglib2-to-shear-an-image/2534/4).

**Comment by ctrueden** [comment]
@LeonYang5114 We can discuss when you get back to LOCI. I would like to add shearing to the Ops `transform` package. We will need this utility method as part of that.

**Comment by hanslovsky** [comment]
As an inspiration, you could have a look at [`BoundingBox`](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/transform/integer/BoundingBox.java) and [`BoundingBoxTransform`](https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/transform/integer/BoundingBoxTransform.java). This class/interface can be easily extended to the `RealTransform` case and could be implemented by any `RealTransform` that transforms the bounding box in a trivial way.

**Comment by bogovicj** [comment]
What should the behavior be for non-linear (e.g. ThinPlateSpline) transforms?  

It could:
- apply the transform on the corners only anyway (with the understanding the the box is an approximation for non-linear xfms)
- apply the transform along many (all?) points along the faces of the source interval 
- throw an exception
- something else?

**Comment by ctrueden** [comment]
For non-linear transforms, we should extend the interface (or provide an optional add-on interface—although with Java 8 default methods, this trick is no longer as necessary for backwards compatibility) so that the transform itself can offer its new bounding box. Of course, with arbitrary transforms the new bounding box could be infinite.

I really want to solve the 99% common cases though, where users want do apply some transform in ImageJ and then have the new image window match the new bounds. Otherwise, even simple things like shearing are not really usable for end users.

**Comment by ctrueden** [comment]
I just ran across [Intervals.smallestContainingInterval(RealInterval)](https://github.com/imglib/imglib2/blob/imglib2-3.2.1/src/main/java/net/imglib2/util/Intervals.java#L268) which is a similar thing.

**Comment by axtimwalde** [comment]
I agree with @hanslovsky, this has to be a complementary interface that is implemented by RealTransforms that can calculate or know their bounding box.

**Comment by ctrueden** [comment]
Thanks @axtimwalde, I agree as well.
